### ISSUE TITLE
[studio] Decryption of data does not honor the new lines in private keys

### ISSUE THREAD

**Issue by AndyReiter** [craftercms/craftercms#5849]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue
The encryption tool does not allow a user to use CDATA sections to preserve values that require whitespace (E.g. PPK/PEM keys)

### Steps to reproduce

Steps:

1. Create a new site
2. Create a new private SSH Key (or use a default example found here: https://phpseclib.com/docs/rsa-keys)
3. Navigate to studio's Encryption Tool and use it to encrypt your private key
4. With the encrypted private key in your clipboard, navigate to Project Tools --> Configuration --> Engine Project Configuration
5. In this configuration file, use the following property and replace the insert-encrypted-key-here with what is stored in your clipboard `<privateKey encrypted="true">insert-encrypted-key-here</privateKey>`
6. Now create a new groovy script under `/scripts/rest` that we will use to print out the key.
7. Use the attached code to print out the key.

### Relevant log output

```shell
Example log output:
https://gist.github.com/AndyReiter/051e3d42d06463c9ba3c5aee7ad46c8d
```

### Screenshots and/or videos

Expected output format:

![image](https://user-images.githubusercontent.com/42217570/199816050-4e7ec72c-f92b-4c7c-b90f-d6ad1a3e40ce.png)

**Comment by russdanner** [comment]
Note this ticket has nothing to do with PPK/PEM keys per se. These are just used as an example of an item someone may want to store as an encrypted config value.

**Comment by sumerjabri** [comment]
In a configuration file, you can try
```
<privateKey encrypted=""><![CDATA[Line 1
Line2]]></privateKey>
```

Run the encryptor and it'll eat the CRLF (new line) and not honor the CDATA. It may also eat the CDATA (not sure).

Once encrypted it looks like this:
```
  <privateKey encrypted="true">${enc:CCE-V1#mkRaRiaLQIMMUW6vqdLeyYDGjf1yw22nuJ1b87vEJhnyaKQwXozR9OCOP1AW81lv}</privateKey>
```

Note that the CDATA section is gone.

What we want it to do is preserve the white space in the CDATA, perhaps it should preserve the CDATA block, or once decrypted it restores it and restores the white space. Please look into the options.

**Comment by phuongnq** [comment]
@sumerjabri I believe the backend (API `/studio/api/2/security/encrypt.json`) is working correctly.

1. Encrypt Marked button in the Engine Configuration screen:

If the text has new lines and we click to encrypt, the text is encrypted correctly since the front-end also sends the newline characters for line breaks:

![2023-01-18_15-23](https://user-images.githubusercontent.com/2996543/213121837-266c17b5-3bfa-4255-95aa-bbc875fd1691.png)


![2023-01-18_15-23_1](https://user-images.githubusercontent.com/2996543/213121967-e3c2a5b3-1600-41f0-8cb9-05767003d54c.png)

Output with the script:
```
println "------------------"
println "private key decrypted:"

def privateKeyContent = siteConfig.getString('privateKey')
println privateKeyContent

return 'OK'
```
![2023-01-18_15-25](https://user-images.githubusercontent.com/2996543/213122216-7236baad-e5a1-4c91-9f44-f1ce9fcce5bf.png)

The same if I tested with a private key (without CDATA tag):

![2023-01-18_15-26](https://user-images.githubusercontent.com/2996543/213122369-2a130c82-f019-4534-8255-57ef9bf0b482.png)
![2023-01-18_15-26_1](https://user-images.githubusercontent.com/2996543/213122400-4242093d-5e7f-4c64-88e7-b9afd4c350be.png)

If the input has a CDATA tag, it also got encrypted since the front end sends the whole as text to the API:
![2023-01-18_15-27](https://user-images.githubusercontent.com/2996543/213122820-f4706d61-3fbf-4ac5-94d7-6d2aacd4e9e7.png)
![2023-01-18_15-28](https://user-images.githubusercontent.com/2996543/213122862-73c05411-5b5b-4ca2-85ac-c57a0905e726.png)

=> In this case, if we want to preserve the CDATA block, the front end should remove it before sending to encrypt API.

2. The global Encryption tool:

Currently, the front end does not support sending a text with newline characters:

![2023-01-18_15-29](https://user-images.githubusercontent.com/2996543/213123253-46fb0c08-ccd4-4f3c-be1b-e425d79d8848.png)

![2023-01-18_15-29_1](https://user-images.githubusercontent.com/2996543/213123364-9b8eb418-1aa8-4a03-8ca4-6fef78b86aa9.png)

But if I use Postman or curl to directly send encryption text with new line, it works properly:

![2023-01-18_15-30](https://user-images.githubusercontent.com/2996543/213123560-b9cab4d2-2145-40b9-be70-677aeeccc14d.png)

![2023-01-18_15-30_1](https://user-images.githubusercontent.com/2996543/213123607-d3b2b321-1249-484d-bd2e-b968a3d1f8b3.png)

In conclusion, the following actions could be fixed from the UI:

* Support CDATA tag while encrypting with *Encrypt Marked*: UI should strip the CDATA block and only send the inside text and keep the CDATA available when writing the configuration with the encrypted data. Ex:
```
<privateKey encrypted="true"><![CDATA[${enc:...}]]></privateKey>
```

* Encryption Tool: Support text with newline characters (CRLF)

Please let me know if we want to discuss more on this.

**Comment by AndyReiter** [comment]
Confirmed fixed in the latest build of 4.0-SNAPSHOT.
